File: .NET/Demo_Common/Entity/DSLEntity.cs
using MessagePack;

namespace Demo_Common.Entity
{
    [MessagePackObject]
    public class DSLEntity
    {
        /// <summary>
        /// Tag ID
        /// </summary>
        [Key(0)]
        public string TagID { get; set; } = "";
        /// <summary>
        /// LED - Red
        /// </summary>
        [Key(1)]
        public bool R { get; set; }
        /// <summary>
        /// LED - Green
        /// </summary>
        [Key(2)]
        public bool G { get; set; }
        /// <summary>
        /// LED - Blue
        /// </summary>
        [Key(3)]
        public bool B { get; set; }
        /// <summary>
        /// Period 闪烁时长, 0~3600, Default 3600
        /// </summary>
        [Key(4)]
        public int Period { get; set; } = 3600;
        /// <summary>
        /// Interval 亮灯间隔, 100~10000, Default 1000
        /// </summary>
        [Key(5)]
        public int Interval { get; set; } = 1000;
        /// <summary>
        /// Duration 亮灯时长, 50~100, Default 50
        /// </summary>
        [Key(6)]
        public int Duration { get; set; } = 50;
        /// <summary>
        /// Token
        /// </summary>
        [Key(7)]
        public int Token { get; set; }
        /// <summary>
        /// 图像数据文件
        /// </summary>
        [Key(8)]
        public byte[] HexData { get; set; } = [];
    }
}

File: Python/Entities/dslEntity.py
import msgpack
from dataclasses import dataclass, field

@dataclass
class DSLEntity:
    """
    DSL Entity for eStation2
    """
    TagID: str = ""
    R: bool = False
    G: bool = False
    B: bool = False
    Period: int = 3600      # -1~3600, Default 3600
    Interval: int = 1000    # 100~10000, Default 1000
    Duration: int = 50      # 50~100, Default 50
    Token: int = 0
    HexData: bytes = field(default_factory=bytes)

    def to_msgpack(self) -> bytes:
        # Serializes the object to MessagePack format using key order
        data = [
            self.TagID,
            self.R,
            self.G,
            self.B,
            self.Period,
            self.Interval,
            self.Duration,
            self.Token,
            self.HexData
        ]
        return msgpack.packb(data)

    @staticmethod
    def from_msgpack(packed: bytes):
        # Deserializes MessagePack data to DSLEntity
        data = msgpack.unpackb(packed)
        return DSLEntity(
            TagID=data[0],
            R=data[1],
            G=data[2],
            B=data[3],
            Period=data[4],
            Interval=data[5],
            Duration=data[6],
            Token=data[7],
            HexData=data[8]
        )